extensions/ql-vscode/src/extension.ts

```diff
@@ -732,6 +733,11 @@ async function activateWithInstalledDistribution(
   );
   ctx.subscriptions.push(databaseUI);
 
+  void extLogger.log("Initializing queries panel.");
```

Comment:
Shall we check whether the relevant feature flag is on before doing the initialization stuff? If we do that there's less of a chance of breaking things while we're in a WIP state, and we avoid logging messages that could confuse the user.

Also, and this can be part of a future PR, we've started using the idea of a "module" (e.g. [`DbModule`](https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/src/databases/db-module.ts)) that deals with initialization etc. and keeps extension.ts changes to a minimum. We could apply the same pattern here.

Reply:
Thanks! I had a go at this in https://github.com/github/vscode-codeql/pull/2418/commits/495b8fc960850fbde026640719c81f826cd67659! (based on the DbModule code)
Let me know if that's the right idea 🙇🏽‍♀️